test/test_sde.py

```diff
@@ -0,0 +1,245 @@
+    elif issubclass(solver_ctr, diffrax.AbstractStratonovichSolver):
+        sol_type = "Stratonovich"
+        ref_solver = diffrax.Heun()
+    else:
+        assert False
+
+    ts_fine = jnp.linspace(t0, t1, 2**14 + 1, endpoint=True)
+    ts_coarse = jnp.linspace(t0, t1, 2**11 + 1, endpoint=True)
+    contr_fine = diffrax.StepTo(ts=ts_fine)
+    contr_coarse = diffrax.StepTo(ts=ts_coarse)
+    saveat = diffrax.SaveAt(ts=jnp.linspace(t0, t1, 2**6 + 1, endpoint=True))
+    levy_area = diffrax.SpaceTimeLevyArea  # must be common for all solvers
+
+    if solutions[sol_type][noise] is None:
+        correct_sol, _ = simple_batch_sde_solve(
+            bmkeys, sde, ref_solver, levy_area, None, contr_fine, 2**-14, saveat
+        )
+        solutions[sol_type][noise] = correct_sol
+    else:
+        correct_sol = solutions[sol_type][noise]
+
+    sol, _ = simple_batch_sde_solve(
+        bmkeys, sde, solver_ctr(), levy_area, None, contr_coarse, 2**-14, saveat
+    )
+    error = path_l2_dist(correct_sol, sol)
+    print(f"Error: {error}")
+    assert error < 0.02
+
+
+def _solvers():
+    yield diffrax.Euler
+    yield diffrax.EulerHeun
+    yield diffrax.Heun
+    yield diffrax.ItoMilstein
+    yield diffrax.Midpoint
+    yield diffrax.ReversibleHeun
+    yield diffrax.StratonovichMilstein
+    yield diffrax.SPaRK
+    yield diffrax.GeneralShARK
+    yield diffrax.SlowRK
+    yield diffrax.ShARK
+    yield diffrax.SRA1
+    yield diffrax.SEA
+
+
+# Define the SDE
+def dict_drift(t, y, args):
+    pytree, _ = args
+    return jtu.tree_map(lambda _, x: -0.5 * x, pytree, y)
+
+
+def dict_diffusion(t, y, args):
+    pytree, additive = args
+
+    def get_matrix(y_leaf):
+        if additive:
+            return 2.0 * jnp.ones(y_leaf.shape + (3,), dtype=jnp.float64)
+        else:
+            return 2.0 * jnp.broadcast_to(
+                jnp.expand_dims(y_leaf, axis=y_leaf.ndim), y_leaf.shape + (3,)
+            )
+
+    return jtu.tree_map(get_matrix, y)
+
+
+@pytest.mark.parametrize("shape", [(), (5, 2)])
+@pytest.mark.parametrize("solver_ctr", _solvers())
+def test_sde_solver_shape(shape, solver_ctr):
+    pytree = ({"a": 0, "b": [0, 0]}, 0, 0)
+    dtype = jnp.float64
+    key = jr.PRNGKey(0)
+    y0 = jtu.tree_map(lambda _: jr.normal(key, shape, dtype=dtype), pytree)
+    t0, t1, dt0 = 0.0, 1.0, 0.3
+
+    # Some solvers only work with additive noise
+    additive = solver_ctr in [diffrax.ShARK, diffrax.SRA1, diffrax.SEA]
+    args = (pytree, additive)
+    solver = solver_ctr()
+    bmkey = jr.PRNGKey(1)
+    struct = jax.ShapeDtypeStruct((3,), dtype)
+    bm_shape = jtu.tree_map(lambda _: struct, pytree)
+    bm = diffrax.VirtualBrownianTree(
+        t0, t1, 0.1, bm_shape, bmkey, diffrax.SpaceTimeLevyArea
+    )
+    terms = MultiTerm(ODETerm(dict_drift), ControlTerm(dict_diffusion, bm))
+    solution = diffrax.diffeqsolve(
+        terms, solver, t0, t1, dt0, y0, args, saveat=diffrax.SaveAt(t1=True)
+    )
+    print(solution.ys)
```

Comment:
Can we remove these print statements? (I assume they're a left-over from debugging?)

Reply:
Oops, removing them now.